src/rsa.rs

```diff
@@ -628,4 +629,37 @@ mod tests {
         assert!(key_pair.sign(&RSA_PKCS1_SHA256, &rng, MESSAGE,
                               &mut signature).is_err());
     }
+
+    // Once the `BN_BLINDING` in an `RSAKeyPair` has been used
+    // `GFp_BN_BLINDING_COUNTER` times, a new blinding should be created.
+    #[cfg(feature = "rsa_signing")]
+    #[test]
+    fn test_signature_rsa_pkcs1_sign_blinding_reuse() {
+        const MESSAGE: &'static [u8] = b"hello, world";
+        let rng = rand::SystemRandom::new();
+
+        const PRIVATE_KEY_DER: &'static [u8] =
+            include_bytes!("signature_rsa_example_private_key.der");
+        let key_bytes_der = untrusted::Input::from(PRIVATE_KEY_DER);
+        let key_pair = RSAKeyPair::from_der(key_bytes_der).unwrap();
+
+        let mut signature = vec![0; key_pair.public_modulus_len()];
+
+        for _ in 0 .. GFp_BN_BLINDING_COUNTER + 1 {
+            let prev_counter = unsafe {
+                let blinding = *(key_pair.blinding.lock().unwrap());
+                (*blinding).counter
```

Comment:
Since the type annotations for `blinding` ended up being unnecessary, we _could_ make this a one-liner, as well as the similar line below. I tried it and found it less readable, but don't feel strongly about it.